lib/archethic/metrics/poller.ex

```diff
@@ -0,0 +1,82 @@
+defmodule ArchEthic.Metrics.Poller do
+  @moduledoc """
+  Provides Telemetry of the network and Maneges
+  """
+  require Logger
+  use GenServer
+  # map because you cant pop from linked list
+  @default_state %{
+    pid_refs: %{},
+    data: ArchEthic.Metrics.Helpers.get_client_metric_default_value()
+  }
+
+  def start_link(_state) do
+    GenServer.start_link(__MODULE__, @default_state, name: __MODULE__)
+  end
+
+  def init(initial_state) do
+    periodic_metric_aggregation()
+    periodic_push_updates()
+    {:ok, initial_state}
+  end
+
+  def periodic_metric_aggregation() do
+    Process.send_after(self(), {:periodic_calculation_of_points}, 5_000)
+  end
+
+  def periodic_push_updates() do
+    Process.send_after(self(), :push_updates, 251)
+  end
+
+  def send_updates(%{data: data, pid_refs: pid_refs}) do
+    Enum.each(pid_refs, fn {pid_k, _pid_v} ->
+      Task.start(fn ->
+        send(pid_k, {:update_data, data})
+      end)
+    end)
+  end
+
+  def monitor() do
+    GenServer.call(__MODULE__, :monitor)
+  end
+
+  def handle_call(:monitor, {pid, __tag}, state) do
+    _mref = Process.monitor(pid)
+    {:reply, :ok, %{state | pid_refs: Map.put(state.pid_refs, pid, nil)}}
```

Comment:
I think you can send the metrics to the clients back with the `send_updates` so a new client will get the latest metrics quickly

Reply:
I agree for the latest updates , but if you can elaborate more on it ,_
As I feel we poll network at 5 seconds , so data will be stale as we are not storing them .

Reply:
Yes we are polling 5 seconds, but stale for 5 seconds for metrics it's fine, as it's the usual Prometheus metrics scrapper interval (so in the industry standard).
Also, the data is stored locally in the GenServer. 
So when the client subscribes, we can send current state as update.

Reply:
Hi , 
I have included that now in latest commit.